scipy/_lib/_array_api.py

```diff
@@ -652,3 +657,115 @@ def xp_default_dtype(xp):
+
+
+
+def _make_capabilities(skip_backends=None, cpu_only=False):
+    skip_backends = [] if skip_backends is None else skip_backends
+
+    capabilities = dataclasses.make_dataclass('capabilities', ['cpu', 'gpu', 'mask'])
+
+    numpy = capabilities(cpu=True, gpu=False, mask=True)
+    strict = capabilities(cpu=True, gpu=False, mask=False)
+    cupy = capabilities(cpu=False, gpu=True, mask=False)
+    torch = capabilities(cpu=True, gpu=True, mask=False)
+    jax = capabilities(cpu=True, gpu=True, mask=False)
+    dask = capabilities(cpu=True, gpu=True, mask=False)
+
+    capabilities = dict(numpy=numpy, array_api_strict=strict, cupy=cupy,
+                        torch=torch, jax=jax, dask=dask)
+
+    for backend, _ in skip_backends:  # ignoring the reason
+        setattr(capabilities[backend], 'cpu', False)
+        setattr(capabilities[backend], 'gpu', False)
+        setattr(capabilities[backend], 'mask', False)
+
+    if cpu_only:
+        cupy.gpu = False
+        torch.gpu = False
+        jax.gpu = False
+        dask.gpu = False
+
+    return capabilities
+
+
+def _make_capabilities_table(capabilities):
+    numpy = capabilities['numpy']
+    cupy = capabilities['cupy']
+    torch = capabilities['torch']
+    jax = capabilities['jax']
+    dask = capabilities['dask']
+
+    for backend in [numpy, cupy, torch, jax, dask]:
+        for attr in ['cpu', 'gpu', 'mask']:
+            val = "✓" if getattr(backend, attr) else " "
+            val += " " * (len('{numpy.}') + len(attr) - 1)
+            setattr(backend, attr, val)
+
+    table = f"""                
+    +---------+-------------+-------------+--------------+
+    | Library | CPU         | GPU         | ``marray``   |
+    +=========+=============+=============+==============+
+    | NumPy   | {numpy.cpu} | {numpy.gpu} | {numpy.mask} |
+    +---------+-------------+-------------+--------------+
+    | CuPy    | {cupy.cpu } | {cupy.gpu } | {cupy.mask } |
+    +---------+-------------+-------------+--------------+
+    | PyTorch | {torch.cpu} | {torch.gpu} | {torch.mask} |
+    +---------+-------------+-------------+--------------+
+    | JAX     | {jax.cpu  } | {jax.gpu  } | {jax.mask  } |
+    +---------+-------------+-------------+--------------+
+    | Dask    | {dask.cpu } | {dask.gpu } | {dask.mask } |
+    +---------+-------------+-------------+--------------+
+    """
+    return table
+
+
+def _make_capabilities_note(fun_name, capabilities):
+    table = _make_capabilities_table(capabilities)
+    note = f"""
+    `{fun_name}` has experimental support for Python Array API Standard compatible
+    backends in addition to NumPy. Please consider testing these features
+    by setting an environment variable ``SCIPY_ARRAY_API=1`` and providing
+    CuPy, PyTorch, JAX, or Dask arrays as array arguments. The following
+    combinations of backend and device (or other capability) are supported.
+    {table}
+     See :ref:`dev-arrayapi` for more information.
```

Comment:
```suggestion
    See :ref:`dev-arrayapi` for more information.
```
Other than this, looks pretty good to me:
![image](https://github.com/user-attachments/assets/d6b27104-748d-4557-894d-e467185ba479)